aws-analytics-pinpoint/src/main/java/com/amplifyframework/analytics/pinpoint/models/PinpointEvent.kt

```diff
@@ -14,32 +14,42 @@
  */
 package com.amplifyframework.analytics.pinpoint.models
 
-import com.google.gson.Gson
 import java.util.UUID
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
 
 /*
 * Internal representation of Pinpoint Event
 * */
+
+@Serializable
 internal class PinpointEvent constructor(
-    private val eventId: String = UUID.randomUUID().toString(),
-    private val eventType: String,
-    private val attributes: Map<String, String>,
-    private val metrics: Map<String, Double>,
-    private val sdkInfo: SDKInfo,
-    private val pinpointSession: PinpointSession,
-    private val eventTimestamp: Long,
-    private val uniqueId: String,
-    private val androidAppDetails: AndroidAppDetails,
-    private val androidDeviceDetails: AndroidDeviceDetails
+    val eventId: String = UUID.randomUUID().toString(),
+    val eventType: String,
+    val attributes: Map<String, String>,
+    val metrics: Map<String, Double>,
+    val sdkInfo: SDKInfo,
+    val pinpointSession: PinpointSession,
+    val eventTimestamp: Long,
+    val uniqueId: String,
+    val androidAppDetails: AndroidAppDetails,
+    val androidDeviceDetails: AndroidDeviceDetails
 ) {
 
     init {
         // TODO("Add documentation")
         require(eventType.length <= 50)
     }
 
+    companion object {
+        fun fromJsonString(jsonString: String): PinpointEvent {
+            return Json.decodeFromString<PinpointEvent>(jsonString)
+        }
+    }
+
     fun toJsonString(): String {
-        val gson = Gson()
-        return gson.toJson(this)
+        return Json.encodeToString(this)
```

Comment:
shouldn't this be autogenerated by `@Serializable` -- something like `Json.encodeToString(PinpointEvent.serializer(), event)`?  or are these for better syntax